std/experimental/checkedint.d

```diff
@@ -916,20 +916,25 @@ if (isIntegral!T || is(T == Checked!(U, H), U, H))
     Returns: A reference to `this`.
     */
     ref Checked opOpAssign(string op, Rhs)(const Rhs rhs) return
-    if (isIntegral!Rhs || isFloatingPoint!Rhs || is(Rhs == bool))
+    if (isIntegral!Rhs || isFloatingPoint!Rhs || is(Rhs == bool) ||
+        is(typeof(Checked!(Rhs, Hook)(rhs))))
     {
-        static assert(is(typeof(mixin("payload" ~ op ~ "=rhs")) == T));
-
-        static if (hasMember!(Hook, "hookOpOpAssign"))
+        static assert(is(typeof(mixin("payload" ~ op ~ "=rhs")) == T) ||
+            is(typeof(Checked!(Rhs, Hook)(rhs))));
+        static if (is(Rhs == Checked!(X, Y), X, Y))
```

Comment:
Why let the compiler do the infersion again?
You already know that its `Checked!(Rhs, Hook)`, no?